plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAdlsConnectorSmokeTest.java

```diff
@@ -58,25 +61,32 @@ public class TestDeltaLakeAdlsConnectorSmokeTest
     private final String container;
     private final String account;
     private final String accessKey;
-    private final BlobContainerClient azureContainerClient;
     private final String adlsDirectory;
+    private BlobContainerClient azureContainerClient;
 
     public TestDeltaLakeAdlsConnectorSmokeTest()
     {
         this.container = requiredNonEmptySystemProperty("testing.azure-abfs-container");
         this.account = requiredNonEmptySystemProperty("testing.azure-abfs-account");
         this.accessKey = requiredNonEmptySystemProperty("testing.azure-abfs-access-key");
-
-        String connectionString = format("DefaultEndpointsProtocol=https;AccountName=%s;AccountKey=%s;EndpointSuffix=core.windows.net", account, accessKey);
-        BlobServiceClient blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionString).buildClient();
-        this.azureContainerClient = blobServiceClient.getBlobContainerClient(container);
         this.adlsDirectory = format("abfs://%s@%s.dfs.core.windows.net/%s/", container, account, bucketName);
     }
 
     @Override
     protected HiveHadoop createHiveHadoop()
             throws Exception
     {
+        String connectionString = format("DefaultEndpointsProtocol=https;AccountName=%s;AccountKey=%s;EndpointSuffix=core.windows.net", account, accessKey);
+        OkHttpClient okHttpClient = new OkHttpClient.Builder().build();
+        closeAfterClass(() -> {
+            okHttpClient.dispatcher().executorService().shutdownNow();
+            okHttpClient.connectionPool().evictAll();
+        });
+        BlobServiceClient blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionString)
+                .httpClient(createAzureHttpClient(okHttpClient, new HttpClientOptions()))
+                .buildClient();
+        this.azureContainerClient = blobServiceClient.getBlobContainerClient(container);
```

Comment:
Why was the creation of the `azureContainerClient` moved out of the constructor?

Reply:
in the constructor i cannot use `closeAfterClass` yet